rbc/tests/test_omnisci_array.py

```diff
@@ -409,3 +409,61 @@ def array_is_null(size):
     _, result = omnisci.sql_execute(query)
 
     assert list(result)[0] == (0,)
+
+
+inps = [('int', 'i4', 'trunc'), ('int', 'i4', 'sext'),
+        ('int', 'i4', 'zext'), ('float', 'f4', 'fptrunc'),
+        ('float', 'f8', 'fpext')]
+
+
+@pytest.mark.parametrize("typ, col, suffix", inps,
+                         ids=[item[-1] for item in inps])
+def test_issue197(omnisci, typ, col, suffix):
+    omnisci.reset()
+
+    import rbc.omnisci_backend as np
+    from numba import types
+
+    @omnisci('int32[](int32[])')
+    def fn_issue197_int_trunc(x):
+        y = np.zeros_like(x)
+        for i in range(len(x)):
+            y[i] = types.int64(x[i] + 3)  # trunc(i64) -> i32
+        return y
+
+    @omnisci('int32[](int32[])')
+    def fn_issue197_int_sext(x):
+        y = np.zeros_like(x)
+        for i in range(len(x)):
+            y[i] = types.int8(x[i] + 3)  # sext(i8) -> i32
+        return y
+
+    @omnisci('int32[](int32[])')
+    def fn_issue197_int_zext(x):
+        y = np.zeros_like(x)
+        for i in range(len(x)):
+            y[i] = types.uint8(x[i] + 3)  # zext(i8) -> i32
+        return y
+
+    @omnisci('float[](float[])')
+    def fn_issue197_float_fptrunc(x):
+        y = np.zeros_like(x)
+        for i in range(len(x)):
+            y[i] = types.float64(x[i] + 3.0)  # fptrunc(f64) -> f3232
+        return y
+
+    @omnisci('double[](double[])')
+    def fn_issue197_float_fpext(x):
+        y = np.zeros_like(x)
+        for i in range(len(x)):
+            y[i] = types.float32(x[i] + 3.0)  # fpext(f32) -> f64
+        return y
```

Comment:
You can join all these functions into one to reduce the compilation burden and code:
```
cast = dict(trunc=types.int64, sext=types.int8, zext=types.uint8, fptrunc=types.float64, fpext=types.float32)[suffix]

@omnisci(f'{typ}[]({typ}[])')
def fn_issue197(x):
    y = np.zeros_like(x)
    for i in range(len(x)):
        y[i] = cast(x[i] + 3)
    return y

fn_name = f"fn_issue197_{typ}_{suffix}"
fn_issue197.__name__ = fn_name

```